File: PYTHON/switchcase2.py
# Fungsi untuk memformat saldo
def format_saldo(saldo):
    return f"Rp {saldo:,}".replace(",", ".")

# Loop untuk meminta nama
nama = ""
konfirmasi = False
while not konfirmasi:
    nama = input("Masukkan nama Anda: ").strip()
    jawab = input("Apakah nama Anda benar? (y/n): ").strip().lower()
    if jawab == 'y':
        konfirmasi = True
    else:
        print("Silakan masukkan nama kembali.")

# Input NIM dan konversi ke saldo
nim = input("Masukkan NIM (tanpa huruf): ").strip()
saldo = int(nim)

# Tampilkan nama dan saldo awal
print(f"\nNama: {nama}")
print(f"Saldo awal: {format_saldo(saldo)}")

# Menu ATM (menggunakan if-elif sebagai equivalent switch-case)
while True:
    print("\n===== MENU ATM =====")
    print("1. Cek Saldo")
    print("2. Tarik Tunai")
    print("3. Setor Tunai")
    print("4. Transfer")
    print("5. Keluar")
    pilihan = input("Pilih menu: ").strip()

    if pilihan == '1':
        print(f"Saldo Anda: {format_saldo(saldo)}")
    elif pilihan == '2':
        jumlah = int(input("Masukkan jumlah tarik tunai: "))
        if jumlah > saldo:
            print("Saldo tidak cukup.")
        else:
            saldo -= jumlah
            print(f"Tarik tunai berhasil. Saldo Anda sekarang: {format_saldo(saldo)}")
    elif pilihan == '3':
        jumlah = int(input("Masukkan jumlah setor tunai: "))
        saldo += jumlah
        print(f"Setor tunai berhasil. Saldo Anda sekarang: {format_saldo(saldo)}")
    elif pilihan == '4':
        rekening = input("Masukkan rekening tujuan: ").strip()
        jumlah = int(input("Masukkan jumlah transfer: "))
        if jumlah > saldo:
            print("Saldo tidak cukup.")
        else:
            saldo -= jumlah
            print(f"Transfer ke {rekening} berhasil. Saldo Anda sekarang: {format_saldo(saldo)}")
    elif pilihan == '5':
        print("Terima kasih telah menggunakan ATM.")
        break
    else:
        print("Pilihan tidak valid.")

File: PHP/switchcase2.php
<?php
// Fungsi untuk memformat saldo
function formatSaldo($saldo) {
    return "Rp " . number_format($saldo, 0, ',', '.');
}

// Input nama
echo "Masukkan nama Anda: ";
$nama = trim(fgets(STDIN));

// Input NIM
echo "Masukkan NIM (tanpa huruf): ";
$nim = trim(fgets(STDIN));
if (!is_numeric($nim) || (int)$nim <= 0) {
    echo "NIM harus angka positif. Program berhenti.\n";
    exit();
}
$saldo = (int)$nim;

// Tampilkan nama dan saldo
echo "\nNama: $nama\n";
echo "Saldo awal: " . formatSaldo($saldo) . "\n";

// Menu ATM
while (true) {
    echo "\n===== MENU ATM =====\n";
    echo "1. Cek Saldo\n";
    echo "2. Tarik Tunai\n";
    echo "3. Setor Tunai\n";
    echo "4. Transfer\n";
    echo "5. Keluar\n";
    echo "Pilih menu: ";
    $pilihan = trim(fgets(STDIN));

    switch ($pilihan) {
        case '1':
            echo "Saldo Anda: " . formatSaldo($saldo) . "\n";
            break;
        case '2':
            echo "Masukkan jumlah tarik tunai: ";
            $jumlah = (int)trim(fgets(STDIN));
            if ($jumlah <= 0) {
                echo "Jumlah harus positif.\n";
            } elseif ($jumlah > $saldo) {
                echo "Saldo tidak cukup.\n";
            } else {
                $saldo -= $jumlah;
                echo "Tarik tunai berhasil. Saldo Anda sekarang: " . formatSaldo($saldo) . "\n";
            }
            break;
        case '3':
            echo "Masukkan jumlah setor tunai: ";
            $jumlah = (int)trim(fgets(STDIN));
            if ($jumlah <= 0) {
                echo "Jumlah harus positif.\n";
            } else {
                $saldo += $jumlah;
                echo "Setor tunai berhasil. Saldo Anda sekarang: " . formatSaldo($saldo) . "\n";
            }
            break;
        case '4':
            echo "Masukkan rekening tujuan: ";
            $rekening = trim(fgets(STDIN));
            if (empty($rekening)) {
                echo "Rekening tujuan tidak boleh kosong.\n";
                break;
            }
            echo "Masukkan jumlah transfer: ";
            $jumlah = (int)trim(fgets(STDIN));
            if ($jumlah <= 0) {
                echo "Jumlah harus positif.\n";
            } elseif ($jumlah > $saldo) {
                echo "Saldo tidak cukup.\n";
            } else {
                $saldo -= $jumlah;
                echo "Transfer ke $rekening berhasil. Saldo Anda sekarang: " . formatSaldo($saldo) . "\n";
            }
            break;
        case '5':
            echo "Terima kasih telah menggunakan ATM.\n";
            exit();
        default:
            echo "Pilihan tidak valid.\n";
    }
}
?>
